### DISCUSSION TITLE
RedirectSection key in doublecmd.xml

### DISCUSSION THREAD

**Discussion by alinaimi** [doublecmd/doublecmd#1321] [Suggestions]
The **RedirectSection** key allows TotalCMD to maintain a light main configuration file by splitting the configuration categories into separate .ini files. Similarly, by implementing this key, DC users can benefit from this key by extracting some categories’ configs and storing them in separate .xml files, which are easier to manage.

**Comment by ags1234** [comment]
This would be interesting to have

**Comment by Skif-off** [comment]
> DC users can benefit from this key by extracting some categories’ configs and storing them in separate .xml files

What categories?

> which are easier to manage.

Example?

P.S. [ADD: Feature [0001289] Store tabs in a separate config file](https://github.com/doublecmd/doublecmd/commit/b38dc095a7640d5f9cb02a487d9fa6f6da59d314).

**Reply by alinaimi** [reply]
I have been using wincmd.ini and doublecmd.xml for a long time, and they have become very large with more than 10k lines. To make it easier to manage them, I use a feature in TC that allows me to divide the wincmd.ini file into smaller .ini files.

i.e., in wincmd.ini, I can specify different .ini files for different configuration sections:
```
[PackerPlugins]
RedirectSection=.config/tc/wcx.ini

[Associations]
RedirectSection=.config/tc/assoc.ini

[Colors]
RedirectSection=.config/tc/themes/**sapphire.ini**

[ColorsDark]
RedirectSection=.config/tc/themes/**sapphire.ini**
```

This works great in TC. This way, I can organize my configuration better and merge them btw machines more easily. It is similar to splitting a big main .c file into smaller .c files and importing them in the main file.

**Comment by alexx2000** [comment]
Colors, associations and tabs already separated.  So doublecmd.xml is much smaller now than before.
